src/models/mSeriesChallenge.ts

```diff
@@ -0,0 +1,287 @@
+import { ListMeta, Model, ModelList } from '../model'
+import { AuthenticatedResponse, SessionHandler } from '../session'
+import { MSeriesChallengeLeaderboardParticipants, MSeriesChallengeLeaderboardResponse, MSeriesChallengeParticipant, MSeriesChallengeParticipantListResponse, MSeriesChallengeParticipantResponse, MSeriesChallengeParticipants, MSeriesChallengeParticipantSorting } from './mSeriesChallengeParticipant'
+
+export const enum MSeriesChallengeType {
+  TimeBased = 'timeBased',
+  GoalBased = 'goalBased',
+}
+
+export const enum MSeriesChallengeFocus {
+  Points = 'points',
+  CaloricBurn = 'caloricBurn',
+  Distance = 'distance',
+  Duration = 'duration',
+}
+
+export const enum MSeriesChallengeRelationship {
+  Owned = 'owned',
+  Joined = 'joined'
+}
+
+export const enum MSeriesChallengeSorting {
+  ID = 'id',
+  StartAt = 'startAt',
+  EndAt = 'endAt',
+  CreatedAt = 'createdAt'
+}
+
+export interface MSeriesChallengeData {
+  id: number
+  joinCode: string
+  name: string
+  userId: number
+  isPublic: boolean
+  userLimit: number | null
+  challengeType: MSeriesChallengeType
+  startAt: string
+  endAt: string | null
+  focus: MSeriesChallengeFocus
+  goal: number
+  isCompleted: boolean
+  isJoined: boolean
+}
+
+export interface MSeriesChallengeResponse extends AuthenticatedResponse {
+  mSeriesChallenge: MSeriesChallengeData
+}
+
+export interface MSeriesChallengeListResponse extends AuthenticatedResponse {
+  mSeriesChallenges: MSeriesChallengeData[]
+  mSeriesChallengesMeta: MSeriesChallengeListResponseMeta
+}
+
+export interface MSeriesChallengeListResponseMeta extends ListMeta {
+  from: string | undefined
+  to: string | undefined
+  isCompleted: boolean
+  relationship: MSeriesChallengeRelationship
+  sort: MSeriesChallengeSorting
+}
+
+export class BaseMSeriesChallenge extends Model {
```

Comment:
No need for the `Base` prefix, but if you are wanting to ensure this is not used you can change it to an `abstract` class.